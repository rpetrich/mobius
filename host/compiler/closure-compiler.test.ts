import closure from "./closure-compiler";
describe("closure", () => {
	it("should name itself", () => {
		expect(closure().name).toBe("closure-compiler");
	});
	it("should compress javascript", () => {
		expect(closure().transformBundle(`window.foo = function(bar) { return bar * bar; }`)).toMatchSnapshot();
	});
});
