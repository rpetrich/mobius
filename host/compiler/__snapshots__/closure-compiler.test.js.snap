// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`closure should compress javascript 1`] = `
Object {
  "code": "window.foo=function(a){return a*a};",
  "map": "{
\\"version\\":3,
\\"file\\":\\"compiled.js\\",
\\"lineCount\\":1,
\\"mappings\\":\\"AAAAA,MAAAC,IAAA,CAAaC,QAAQ,CAACC,CAAD,CAAM,CAAE,MAAOA,EAAP,CAAaA,CAAf;\\",
\\"sources\\":[\\"Input_0\\"],
\\"names\\":[\\"window\\",\\"foo\\",\\"window.foo\\",\\"bar\\"]
}
",
}
`;
